sys/kern/kern_proc.c

```diff
@@ -2605,8 +2605,9 @@ sysctl_kern_proc_c18n_compartments(SYSCTL_HANDLER_ARGS)
 	struct proc *p;
 	struct cheri_c18n_info info;
 	struct cheri_c18n_compart compart;
+	char * __capability entryp;
```

Comment:
I would declare a structure that you read in at once of both the pointer and bool.  Or even better, perhaps use the model of link_map where you declare the "common" structure in a header in sys/sys and embed that structure as a member in the structure used in rtld in userspace.  You can then read in the entire structure here in the kernel in one go.  If you wanted to follow the normal form here, you would maybe use `struct cheri_c18n_compart` to be the link_map-like thing and then rename your existing `cheri_c18n_compart` to be `struct kinfo_cheri_c18n_compart`.  In particular, this arrangement will prove much easier to extend in the future.

Reply:
I've implemented the link-map-like structure, but am still using an array to store them.